app.ts

```diff
@@ -0,0 +1,323 @@
+				th.textContent = columnName;
+				tableHeaderRow.append(th);
+			}
+			const recordCount = await totalRecords();
+			totalPages = Math.ceil(recordCount / state.recordsPerPage);
+			updateScreen();
+		}).catch((error) => {
+			throw error;
+		})
+}
+
+// Fetch a range of records from the server based on specified indices.
+function fetchRecords(fromRecord: number, toRecord: number): Promise<any> {
+	if (fromRecord > toRecord) {
+		fromRecord = toRecord
+	}
+	return fetch(`${state.IMQS}/records?from=${fromRecord}&to=${toRecord}`)
+		.then((response) => response.text())
+		.then((dataText) => JSON.parse(dataText))
+		.catch((error) => {
+			throw error;
+		})
+}
+
+// Calculate the number of records that can fit on the screen.
+function screenCalculations(screenHeight: number): number {
+	const estimatedRowHeightFactor = 1;
+	const estimatedRowHeight = estimatedRowHeightFactor * 50;
+	const recordsPerPage = Math.floor(screenHeight / estimatedRowHeight) - 1;
+	return Math.max(recordsPerPage - 2, 1);
+}
+
+// Handles the input for the search
+function inputHandling(): Promise<void> {
+	return totalRecords()
+		.then((totalRecCount) => {
+			const searchInput = document.getElementById("searchInput") as HTMLInputElement;
+			searchInput.max = (totalRecCount - 1).toString();
+		})
+}
+
+function debounce(func: any, delay: number) {
+	let timeoutId: any;
+	return function (...args: any) {
+		clearTimeout(timeoutId);
+		timeoutId = setTimeout(() => {
+			func(...args);
+		}, delay);
+	}
+}
+
+// Display records on the page based on the specified range.
+async function displayData(fromRecord: number, recordsDisplayed: number) {
+	$("#loader").show();
+	$("#tableWrapper").hide();
+	let isSearchMode = state.searchedIndex !== null;
+	const searchIndexOnPage = isSearchMode ? state.searchedIndex! - fromRecord : 0;
+	fromRecord = Math.max(fromRecord, 1);
+	const recordCount = await totalRecords();
+	const maximumRecords = Math.min(recordCount - fromRecord, recordsDisplayed);
+	const toRecord = fromRecord + maximumRecords - 1;
+	const data = await fetchRecords(fromRecord, toRecord);
+	let tableData = "";
+	if (isSearchMode) {
+		if (currentPage * state.recordsPerPage > recordCount) {
+			recordsDisplayed = recordCount - fromRecord;
+		}
+		state.currentValueOfFirstRecord = parseInt(data[0][0]);
+		state.currentFirstRecordIndex = fromRecord;
+		if (recordsDisplayed >= 16) {
+			currentPage = totalPages;
+			state.currentFirstRecordIndex = (totalPages - 1) * state.recordsPerPage;
+		}
+	} else {
+		fromRecord = Math.max(
+			fromRecord + searchIndexOnPage - (recordsDisplayed - 1), 0);
+	}
+	for (const record of data) {
+		tableData += "<tr>";
+		for (const value of record) {
+			tableData += `<td>${value}</td>`;
+		}
+		tableData += "</tr>";
+	}
+	$("#tableBody").html(tableData);
+	$("#loader").hide();
+	$("#tableWrapper").show();
+}
+
+// Search for records by a given value and display them on the page.
+async function searchMethod(searchValue: any) {
```

Comment:
```suggestion
async searchMethod(searchValue: number): Promise<fill in your return type> {
```
Give return types and parameter type.